src/api/api.cc

```diff
@@ -10925,6 +10927,14 @@ void RecordReplayTriggerProgressInterrupt() {
 
 void RecordReplayOnTargetProgressReached() {
   CHECK(IsMainThread());
+  if (recordreplay::AreEventsDisallowed()) {
+    // We should not have progress updates when events disallowed.
+    if (!recordreplay::HasDivergedFromRecording()) {
+        recordreplay::Warning("OnProgressReached %s",
```

Comment:
Note: The `EventsDisallowed` label will be appended to the message automatically.